homeassistant/components/mqtt/device_trigger.py

```diff
@@ -239,22 +254,41 @@ async def async_setup(self) -> None:
     async def async_update(self, discovery_data: MQTTDiscoveryPayload) -> None:
         """Handle MQTT device trigger discovery updates."""
         discovery_hash = self.discovery_data[ATTR_DISCOVERY_HASH]
-        discovery_id = discovery_hash[1]
         debug_info.update_trigger_discovery_data(
             self.hass, discovery_hash, discovery_data
         )
         config = TRIGGER_DISCOVERY_SCHEMA(discovery_data)
+        if (
+            new_trigger_id
+            := f"{self.device_id}_{config[CONF_TYPE]}_{config[CONF_SUBTYPE]}"
+        ) != self.trigger_id:
```

Comment:
This sure aint pretty after it's been mangled by Black, I think this looks better:
```suggestion
        new_trigger_id = f"{self.device_id}_{config[CONF_TYPE]}_{config[CONF_SUBTYPE]}"
        if new_trigger_id != self.trigger_id:
```